cli/cmd/setup.go

```diff
@@ -29,6 +31,88 @@ func setupOCMConfig(cmd *cobra.Command) {
 	}
 }
 
+func setupFilesystemConfig(cmd *cobra.Command) {
+	tempFolderFlag, _ := cmd.PersistentFlags().GetString("temp-folder")
+
+	ocmCtx := ocmctx.FromContext(cmd.Context())
+	cfg := ocmCtx.Configuration()
+
+	var fsCfg *v1alpha1.Config
+	var err error
+
+	if cfg == nil {
+		slog.WarnContext(cmd.Context(), "could not get configuration to initialize filesystem config")
+		fsCfg = &v1alpha1.Config{}
+	} else {
+		fsCfg, err = v1alpha1.LookupConfig(cfg)
+		if err != nil {
+			slog.WarnContext(cmd.Context(), "could not get filesystem configuration", slog.String("error", err.Error()))
+			fsCfg = &v1alpha1.Config{}
+		}
+	}
+
+	// CLI flag takes precedence over config-file
+	if tempFolderFlag != "" {
+		fsCfg.TempFolder = tempFolderFlag
+
+		// If we have a CLI flag but no filesystem config in the config,
+		// we need to add it to the configuration
+		if cfg != nil && !hasFilesystemConfig(cfg) {
+			if err := addFilesystemConfigToCentralConfig(cmd, fsCfg); err != nil {
+				slog.WarnContext(cmd.Context(), "could not add filesystem config to central configuration", slog.String("error", err.Error()))
+			}
+		}
+	}
+
+	ctx := ocmctx.WithFilesystemConfig(cmd.Context(), fsCfg)
+	cmd.SetContext(ctx)
+}
+
+// hasFilesystemConfig checks if the central configuration already contains filesystem configuration
+func hasFilesystemConfig(cfg *v1.Config) bool {
+	if cfg == nil {
+		return false
+	}
+	for _, configEntry := range cfg.Configurations {
+		if configEntry.Name == v1alpha1.ConfigType {
+			return true
+		}
+	}
+
+	return false
+}
+
+// addFilesystemConfigToCentralConfig adds the filesystem configuration to the central configuration
+func addFilesystemConfigToCentralConfig(cmd *cobra.Command, fsCfg *v1alpha1.Config) error {
+	ocmCtx := ocmctx.FromContext(cmd.Context())
+	cfg := ocmCtx.Configuration()
+	if cfg == nil {
+		return fmt.Errorf("no central configuration available")
+	}
+
+	// Set the type for the filesystem config
+	fsCfg.Type = runtime.NewVersionedType(v1alpha1.ConfigType, v1alpha1.Version)
+
+	// Create a runtime.Raw directly with the filesystem config
+	raw := &runtime.Raw{
+		Type: fsCfg.Type,
+	}
+
+	// Use the v1 scheme to encode the filesystem config
+	if err := v1.Scheme.Convert(fsCfg, raw); err != nil {
```

Comment:
why do you not just initialize the config directly?

Reply:
Ah, something like this would work, right?
```go
	raw := &runtime.Raw{}
	if err := v1.Scheme.Convert(fsCfg, raw); err != nil {
		return fmt.Errorf("failed to convert filesystem config to raw: %w", err)
	}
	cfg.Configurations = append(cfg.Configurations, raw)
```

Since config already should have type set through the scheme?

Reply:
Yah, it appears to be working... Tests are passing at least. :D